File: tutorials/testing/c++/graph_map_test.cpp
#include <string>
#include "Bridges.h"
#include "DataSource.h"
#include "GraphAdjList.h"

using namespace bridges;

int main(int argc, char **argv) {
	// create Bridges object
    Bridges bridges (YOUR_ASSSIGNMENT_NUMBER, "YOUR_USER_ID",
        "YOUR_API_KEY");

	// create an adjacency list based graph
	GraphAdjList<string, string> g;

	for (int i = 0; i < 10; i++) {
		g.addVertex("node" + std::to_string(i), "node" + std::to_string(i));
		if (i > 0) {
			g.addEdge("node" + std::to_string(i), "node" + std::to_string(i - 1));
		}
	}

	// Pass the graph object to BRIDGES
	bridges.setDataStructure(g);

	// visualize SVG AlbersUSA map
	bridges.setCoordSystemType("albersusa");
	bridges.setMapOverlay(true);
	bridges.visualize();

	// visualize SVG World map
	bridges.setCoordSystemType("equirectangular");
	bridges.setMapOverlay(true);
	bridges.setTitle("SVG World Map");
	bridges.visualize();


	// CANVAS VERSIONS (add >100 nodes)
	for (int i = 10; i < 110; i++) {
		g.addVertex("node" + std::to_string(i), "node" + std::to_string(i));
		if (i > 0) {
			g.addEdge("node" + std::to_string(i), "node" + std::to_string(i - 1));
		}
	}

	// fix a node over Charlotte, NC
	g.getVertex("node6")->setLocation(-80.8431, 35.2271);
	g.getVertex("node6")->setColor("red");

	// visualize Canvas AlbersUSA map
	bridges.setCoordSystemType("albersusa");
	bridges.setMapOverlay(true);
	bridges.setTitle("Canvas USA Map");
	bridges.visualize();

	// fix a node over London
	g.getVertex("node10")->setLocation(-0.1278, 51.5074);
	g.getVertex("node10")->setColor("red");

	// visualize Canvas World map
	bridges.setCoordSystemType("equirectangular");
	bridges.setMapOverlay(true);
	bridges.setTitle("Canvas World Map");
	bridges.visualize();
}

File: tutorials/testing/python/graph_map_test.py
from bridges.data_src_dependent import *
from bridges.graph_adj_list import *
from bridges.bridges import *
import sys


def main():

    # create the Bridges object, set credentials
    bridges = Bridges(YOUR_ASSSIGNMENT_NUMBER, "YOUR_USER_ID", "YOUR_API_KEY")

    # title, description
    bridges.title = "Testing map capabilities in BRIDGES with graphs"

    # create an adjacency list based graph
    g = GraphAdjList()

    # create some nodes
    for i in range(10):
       g.add_vertex("node" + str(i), "node" + str(i))

    for i in range(10):
       if i > 0: 
          g.add_edge("node" + str(i), "node" + str(i - 1))


    #visualize in SVG AlbersUSA map
    bridges.set_coord_system_type("albersusa")
    bridges.set_map_overlay(True)

    bridges.set_data_structure(g)
    bridges.visualize()


    # visualize SVG World map
    bridges.set_coord_system_type("equirectangular")
    bridges.set_map_overlay(True)
    bridges.set_title("SVG World Map")
    bridges.visualize()

    # CANVAS VERSIONS (add >100 nodes)
    for i in range(110):
        g.add_vertex("node" + str(i), "node" + str(i))
        if i > 0:
            g.add_edge("node" + str(i), "node" + str(i - 1))

    #fix a node over Charlotte, NC
    g.get_vertex("node6").set_location(-80.8431, 35.2271)
    g.get_vertex("node6").color = "red"

    # visualize Canvas AlbersUSA map
    bridges.set_coord_system_type("equirectangular")
    bridges.set_map_overlay(True)
    bridges.title = "Canvas USA Map"
    bridges.visualize()

    #fix a node over London
    g.get_vertex("node10").set_location(-0.1278, 51.5074)
    g.get_vertex("node10").color = "red"

    # visualize Canvas World map
    bridges.set_coord_system_type("equirectangular")
    bridges.set_map_overlay(True)
    bridges.title = "Canvas World Map"
    bridges.visualize()


if __name__ == "__main__":
    main()
